src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -158,35 +160,56 @@ private class RsLiteralWebReferenceProvider : PsiReferenceProvider() {
 
     override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<PsiReference> {
         val stringLiteral = (element as? RsLitExpr)?.kind as? RsLiteralKind.String ?: return PsiReference.EMPTY_ARRAY
-        // Url should contain `:` symbol.
-        // In combination with the fact that `textContains` doesn't allocate memory,
-        // it makes quite cheap check
-        if (!element.textContains(':')) return PsiReference.EMPTY_ARRAY
-
-        val valueRange = stringLiteral.offsets.value ?: return PsiReference.EMPTY_ARRAY
-        val rawValue = stringLiteral.rawValue ?: return PsiReference.EMPTY_ARRAY
-        val (content, indexFn) = if (stringLiteral.node.elementType in RS_RAW_LITERALS) {
-            rawValue to { i: Int -> i }
-        } else {
-            val (content, indices) = parseRustStringCharacters(rawValue)
-            content to { i: Int -> indices[i] }
-        }
-
-        val references = SmartList<PsiReference>()
-        var index = 0
-        for (word in content.split(SPACE_SPLITTER)) {
-            if (word.isNotEmpty() && GlobalPathReferenceProvider.isWebReferenceUrl(word)) {
-                val startOffset = valueRange.startOffset + indexFn(index)
-                val endOffset = valueRange.startOffset + indexFn(index + word.length)
-                references += WebReference(element, TextRange(startOffset, endOffset), word)
-            }
-            index += word.length + 1
-        }
 
-        return references.toArray(PsiReference.EMPTY_ARRAY)
+        return ifUrl(element, stringLiteral)
+            ?: ifRustGitHubIssue(element, stringLiteral)
+            ?: PsiReference.EMPTY_ARRAY
```

Comment:
Actually, I don't like that we merge two independent cases into single `RsLiteralWebReferenceProvider`

Let's better provide two separate providers. If you don't like to duplicate some code, you can provide base class like the following:
```kotlin
private abstract class RsLiteralWebReferenceProviderBase() : PsiReferenceProvider() {

    // Web references do not point to any real PsiElement
    override fun acceptsTarget(target: PsiElement): Boolean = false

    final override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<PsiReference> {
        val stringLiteral = (element as? RsLitExpr)?.kind as? RsLiteralKind.String ?: return PsiReference.EMPTY_ARRAY
        return getReferencesByLiteral(element, stringLiteral)
    }

    protected abstract fun getReferencesByLiteral(element: PsiElement, stringLiteral: RsLiteralKind.String): Array<PsiReference>
}
```